cmd/xamarin.go

```diff
@@ -87,14 +88,38 @@ func scanXamarinProject(cmd *cobra.Command, args []string) error {
 	// Xamarin Solution Path
 	xamarinCmd.SolutionFilePath = paramXamarinSolutionFilePath
 	if xamarinCmd.SolutionFilePath == "" {
-		askText := `Please drag-and-drop your Xamarin Solution (` + colorstring.Green(".sln") + `) file,
-and then hit Enter`
+		var solutionPth string
+
 		fmt.Println()
-		projpth, err := goinp.AskForPath(askText)
+		log.Infof("Scan the directory for solution files")
+		log.Warnf("You can specify the Xamarin Solution file to scan with the --file flag.")
+
+		solPaths, err := scanForProjectFiles(xamarinIDE)
 		if err != nil {
-			return fmt.Errorf("failed to read input: %s", err)
+			log.Printf("Failed: %s", err)
+			fmt.Println()
+
+			log.Infof("Provide the solution file manually")
+			askText := `Please drag-and-drop your Xamarin Solution (` + colorstring.Green(".sln") + `) file,
+and then hit Enter`
+			solutionPth, err = goinp.AskForPath(askText)
+			if err != nil {
+				return fmt.Errorf("failed to read input: %s", err)
+			}
+		} else {
```

Comment:
Would move this block to a function, in addition with returning early instead of using else extensively.

Reply:
Fixed in:
https://github.com/bitrise-tools/codesigndoc/pull/95/commits/2ab734ef0a19e800cc4ef305a61d03cb5ea2ed3d

---

Do the same for Xcode as well:
https://github.com/bitrise-tools/codesigndoc/pull/95/commits/f8b95cad7b28baafc1c3fa34a93b33e00a93d85b